app/models/constraint.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class Constraint < ApplicationRecord
+  validates :category, :name, presence: true
+  validates :name, uniqueness: true
+
+  enum category: {
+    flooding: "flooding",
+    military_and_defence: "military_and_defence",
+    ecology: "ecology",
+    heritage_and_conservation: "heritage_and_conservation",
+    general_policy: "general_policy",
+    tree: "tree",
+    other: "other",
+    custom: "custom"
+  }
+
+  has_many :planning_application_constraints, dependent: :destroy
+end
```

Comment:
Add
```
validates :name, uniqueness: {scope: :local_authority}

belongs_to :local_authority, optional: true
```

Also change `custom` to `local`

Reply:
On `local_authority` add `has_many :constraints`

Reply:
☝🏻 this came out of a discussion with Andrew. We've realised that custom constraint ought to be scoped to the local authority, so that e.g. bucks and lambeth could both have a constraint with the same name, but without allowing a local authority to have multiple constraints with the same name, and only showing them their own custom constraints.

Reply:
Yeah had this in mind for the subsequent PR as I didn't have time to finish but thanks for adding here